internal/encryption/keyprovider/openbao/config.go

```diff
@@ -0,0 +1,97 @@
+package openbao
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	openbao "github.com/openbao/openbao/api"
+	"github.com/opentofu/opentofu/internal/encryption/keyprovider"
+)
+
+type Config struct {
+	Address string `hcl:"address,optional"`
+	Token   string `hcl:"token,optional"`
+
+	KeyName   string `hcl:"key_name"`
+	KeyLength int    `hcl:"key_length,optional"`
+}
+
+func (c Config) Build() (keyprovider.KeyProvider, keyprovider.KeyMeta, error) {
+	if c.KeyName == "" {
```

Comment:
Unfortunately, there are no formatting rules in public docs. I suggest relying on server-side validation since these rules could differ slightly for multiple OpenBao-compatible implementations.

Reply:
In that case, please note the urlencode comments below.